doc/hardware/peripherals/retained_mem.rst

```diff
@@ -0,0 +1,24 @@
+.. _retained_mem_api:
+
+Retained memory
+###############
+
+Overview
+********
+
+The retained memory driver API provides a way of reading from/writing to memory
+areas whereby the contents of the memory is retained whilst the device is
+powered (data may be lost in low power modes).
```

Comment:
> to memory areas whereby the contents of the memory is retained whilst the device is
powered (data may be lost in low power modes).

Sorry to come back on this, but description just matches plain usual RAM, which doesn't exactly fit the notion of retention.. Any way to make it clearer ?

Reply:
Not sure how to clarify that, for example on nordic parts, GPREGRET is 1 or 2 registers (an area), the RAM driver would be an area, not sure about any other devices that would add a driver for this.

Reply:
Didn't you mention about persistence across resets ?
Is that a required property for those areas ?

Reply:
Resets yes: `whereby the contents of the memory is retained whilst the device is powered` i.e. when the device remains powered but the application reboots, not if power to the device is lost